test/state/mpt.cpp

```diff
@@ -237,7 +244,7 @@ bytes MPTNode::encode() const  // NOLINT(misc-no-recursion)
         for (const auto& child : m_children)
         {
             if (child)
-                encoded += rlp::encode(keccak256(child->encode()));
+                encoded += encode_child(*child);
```

Comment:
Noy sure why lambda is needed, just to avoid +1 nesting level?

Reply:
Okay if you wish, but I kinda jump up and down the function to understand it :upside_down_face:

Reply:
I needed a helper function because this is used for `branch` and `ext` nodes. It can be a regular static function. I was just curious how lambda works in this case.

Reply:
I ah see, didn't noticed it's reused in ext.